spec/features/sign_up_spec.rb

```diff
@@ -0,0 +1,25 @@
+require 'rails_helper'
+
+RSpec.feature 'User sign up', type: :feature do
+  let(:email) { Faker::Internet.email }
+  let(:password) { Faker::Internet.password(min_length: 6) }
+  before { visit new_user_registration_path }
+
+  it 'with valid data user should registered' do
+    fill_in 'Email', with: email
+    fill_in 'Password', with: password
+    fill_in 'Password confirmation', with: password
+    click_button 'Sign up'
+    expect(page).to have_content 'Welcome! You have signed up successfully.'
+    expect(page).to have_current_path root_path
+  end
+
+  it 'when email already exists user should not registered' do
+    user = create(:user)
+    fill_in 'Email', with: user.email
+    fill_in 'Password', with: user.password
+    fill_in 'Password confirmation', with: user.password
+    click_button 'Sign up'
+    expect(page).to have_content 'Email has already been taken'
+  end
+end
```

Comment:
We can add more cases , like without email, without password, with wrong password confirmation etc.